WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsViewController.m

```diff
@@ -855,9 +855,6 @@ - (void)navigateToCommentIDIfNeeded
             [self.tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionTop animated:YES];
             self.highlightedIndexPath = indexPath;
         }
-
-        // Reset the commentID so we don't do this again.
-        self.navigateToCommentID = nil;
```

Comment:
How does removing this code cause the scrollview to scroll to the right comment, I am bit confused here.

Reply:
When we enter the screen, we will see a list of the first comment and a highlighted comment. Right after the screen is presented, we fetch all the comments (which might be a few pages) until the response contains the highlighted post. The problem is that after loading each page, it tries to scroll down to the highlighted comment **only once** after loading the first page, and while scrolling down, the app loads more pages that push the highlighted comment's position. As a result, the comment shown after scrolling down needs to be corrected.
I observed this while debugging it. The entire pagination/scroll mechanism is too tricky for me to modify. At least not every attempt was successful.
That's why I just let navigate to the comment after every page is fetched to guarantee the user sees the highlighted comment. Although it looks pretty clumsy, it could be a good starting point unless we refactor this entire screen one day (during Notification Phase 3, for example, 😬.)

Reply:
I see. These are not easy nuts to crack. It is though to choose a glitchy working scroll versus a broken smooth scroll :D But if that line is not expected to cause other regression, I think we can keep this solution as an improvement as you stated.